contracts/oracles/WstEthOracle.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: bsl-1.1
+
+pragma solidity 0.7.6;
+
+import "../helpers/SafeMath.sol";
+import "../helpers/ERC20Like.sol";
+import "../interfaces/IWstEthToken.sol";
+import "../interfaces/IStEthPriceFeed.sol";
+import "../interfaces/IOracleUsd.sol";
+import "../interfaces/IOracleRegistry.sol";
+import "../interfaces/IOracleEth.sol";
+import "../VaultParameters.sol";
+
+/**
+ * @title WstEthOracle
+ * @dev Wrapper to quote wstETH ERC20 token that represents the account's share of the total supply of stETH tokens. https://docs.lido.fi/contracts/wsteth/
+ **/
+
+contract WstEthOracle is IOracleUsd, Auth  {
+    using SafeMath for uint;
+
+    IOracleRegistry public immutable oracleRegistry;
+
+    address public stEthPriceFeed;
+
+    uint immutable stEthDecimals;
+
+    address public immutable addressWETH;
+
+    uint constant public MAX_SAFE_PRICE_DIFF = 500;
+
+    event StEthPriceFeedChanged(address indexed implementation);
+
+    constructor(address _vaultParameters, address _oracleRegistry, address _wstETH, address _stETHPriceFeed) Auth(_vaultParameters) {
+        require(_vaultParameters != address(0) && _oracleRegistry != address(0) && _wstETH != address(0) && _stETHPriceFeed != address(0), "Unit Protocol: ZERO_ADDRESS");
+        oracleRegistry = IOracleRegistry(_oracleRegistry);
+        addressWETH = IOracleRegistry(_oracleRegistry).WETH();
+        stEthPriceFeed = _stETHPriceFeed;
+        address stEthToken = IWstEthToken(_wstETH).stETH();
+        require(stEthToken != address(0), "Unit Protocol: ZERO_ADDRESS");
+        stEthDecimals = ERC20Like(stEthToken).decimals();
+    }
+
+    function setStEthPriceFeed(address impl) external onlyManager {
+      require(impl != address(0), "Unit Protocol: ZERO_ADDRESS");
+      stEthPriceFeed = impl;
+      emit StEthPriceFeedChanged(impl);
+    }
+
+    function getDecimalsStEth() public view returns (uint) {
+        return stEthDecimals;
+    }
+
+    function _percentage_diff(uint nv, uint ov) private view returns (uint) {
+        if (nv > ov) {
+          return ( nv - ov ) * 10000 / ov;
+        } else {
+          return ( ov - nv ) * 10000 / ov;
+        }
+    }
+
+    function has_changed_unsafely(uint256 pool_price, uint256 oracle_price) private view returns (bool) {
+        return _percentage_diff(pool_price, oracle_price) > MAX_SAFE_PRICE_DIFF;
+    }
+
+    // returns Q112-encoded value
+    function assetToUsd(address bearing, uint amount) public override view returns (uint) {
+        if (amount == 0) return 0;
+        uint _qtyStEthByWstEth = IWstEthToken(bearing).getStETHByWstETH(amount);
```

Comment:
We assume `stEthToken = IWstEthToken(_wstETH).stETH()` in the ctor, however, here we accept any `bearing`, potentially different from `_wstETH`.

Reply:
+